### ISSUE TITLE
MPV based media players exits with "VA error: wayland: Wayland roundtrip error: Protocol error (errno 71)" on GNOME/Wayland

### ISSUE THREAD

**Issue by bbhtt** [flatpak/flatpak#4289]
## Linux distribution and version

Arch Linux/GNOME (shell 40.1)

wayland 1.19

## Flatpak version

flatpak 1.11.1

## Description of the problem

Lots of MPV or libmpv based players utilizing hardware acceleration on Wayland exits with: 

```
wl_drm@23: error 0: authenticate failed
VA error: wayland: Wayland roundtrip error: Protocol error (errno 71)
```

At the moment:

1) Celluloid
2) MPV 
3) Jellyfin https://github.com/jellyfin/jellyfin-media-player/issues/90
4) Haruna (maybe ? Not sure how to check)

from Flathub seems affected.

**Distribution version of Celluloid or MPV running under Wayland + vaapi works fine**:

`GDK_BACKEND=wayland mpv --hwdec=vaapi --gpu-context=wayland *.mkv`

## Steps to reproduce

`$ flatpak run io.mpv.Mpv --hwdec=vaapi --gpu-context=wayland *.mkv`

```
[mpv_thumbnail_script_client_osc]  
[mpv_thumbnail_script_client_osc] The 'tracks-changed' event is deprecated and will be removed.
wl_drm@23: error 0: authenticate failed
VA error: wayland: Wayland roundtrip error: Protocol error (errno 71)
[vaapi] libva: vaGetDriverNameByIndex() failed with invalid VADisplay, driver_name = (null)
[vo/gpu/wayland] Error occurred on the display fd, closing
```

Adding `--env=WAYLAND_DEBUG=1`

```
[351076.016]  -> wl_display@1.get_registry(new id wl_registry@2)
[351076.237]  -> wl_display@1.sync(new id wl_callback@3)
[351076.634] wl_display@1.delete_id(3)
[351076.775] wl_registry@2.global(1, "wl_drm", 2)
[351077.041] wl_registry@2.global(2, "wl_compositor", 4)
[351077.416]  -> wl_registry@2.bind(2, "wl_compositor", 3, new id [unknown]@4)
[351077.789]  -> wl_compositor@4.create_surface(new id wl_surface@5)
[351077.930]  -> wl_compositor@4.create_surface(new id wl_surface@6)
[351078.183] wl_registry@2.global(3, "wl_shm", 1)
[351078.439]  -> wl_registry@2.bind(3, "wl_shm", 1, new id [unknown]@7)
[351078.974] wl_registry@2.global(4, "wl_output", 2)
[351079.344]  -> wl_registry@2.bind(4, "wl_output", 2, new id [unknown]@8)
[351079.732] wl_registry@2.global(5, "zxdg_output_manager_v1", 3)
[351080.111] wl_registry@2.global(6, "wl_data_device_manager", 3)
[351080.478]  -> wl_registry@2.bind(6, "wl_data_device_manager", 3, new id [unknown]@9)
[351080.748] wl_registry@2.global(7, "zwp_primary_selection_device_manager_v1", 1)
[351081.115] wl_registry@2.global(8, "gtk_primary_selection_device_manager", 1)
[351081.371] wl_registry@2.global(9, "wl_subcompositor", 1)
[351081.631] wl_registry@2.global(10, "xdg_wm_base", 3)
[351082.004]  -> wl_registry@2.bind(10, "xdg_wm_base", 2, new id [unknown]@10)
[351082.497] wl_registry@2.global(11, "zxdg_shell_v6", 1)
[351082.759] wl_registry@2.global(12, "wl_shell", 1)
[351083.128] wl_registry@2.global(13, "gtk_shell1", 4)
[351083.414] wl_registry@2.global(14, "wp_viewporter", 1)
[351083.784] wl_registry@2.global(15, "zwp_pointer_gestures_v1", 1)
[351084.044] wl_registry@2.global(16, "zwp_tablet_manager_v2", 1)
[351084.301] wl_registry@2.global(17, "wl_seat", 5)
[351084.565]  -> wl_registry@2.bind(17, "wl_seat", 1, new id [unknown]@11)
[351084.975] wl_registry@2.global(18, "zwp_relative_pointer_manager_v1", 1)
[351087.891] wl_registry@2.global(19, "zwp_pointer_constraints_v1", 1)
[351088.159] wl_registry@2.global(20, "zxdg_exporter_v1", 1)
[351088.414] wl_registry@2.global(21, "zxdg_importer_v1", 1)
[351088.449] wl_registry@2.global(22, "zwp_linux_dmabuf_v1", 3)
[351088.473] wl_registry@2.global(23, "zwp_keyboard_shortcuts_inhibit_manager_v1", 1)
[351088.502] wl_registry@2.global(24, "zwp_text_input_manager_v3", 1)
[351088.528] wl_registry@2.global(25, "gtk_text_input_manager", 1)
[351088.554] wl_registry@2.global(26, "wp_presentation", 1)
[351088.586]  -> wl_registry@2.bind(26, "wp_presentation", 1, new id [unknown]@12)
[351088.626] wl_callback@3.done(3802)
[351088.643]  -> xdg_wm_base@10.get_xdg_surface(new id xdg_surface@3, wl_surface@5)
[351088.668]  -> xdg_surface@3.get_toplevel(new id xdg_toplevel@13)
[351088.697]  -> xdg_toplevel@13.set_app_id("mpv")
[351088.739]  -> wl_data_device_manager@9.get_data_device(new id wl_data_device@14, wl_seat@11)
[351091.491]  -> wl_display@1.get_registry(new id wl_registry@15)
[351091.575]  -> wl_display@1.sync(new id wl_callback@16)
[351109.718] wl_display@1.delete_id(16)
[351109.767] wl_registry@15.global(1, "wl_drm", 2)
[351109.803]  -> wl_registry@15.bind(1, "wl_drm", 2, new id [unknown]@17)
[351109.865] wl_registry@15.global(2, "wl_compositor", 4)
[351109.893] wl_registry@15.global(3, "wl_shm", 1)
[351109.921] wl_registry@15.global(4, "wl_output", 2)
[351109.948] wl_registry@15.global(5, "zxdg_output_manager_v1", 3)
[351109.974] wl_registry@15.global(6, "wl_data_device_manager", 3)
[351110.001] wl_registry@15.global(7, "zwp_primary_selection_device_manager_v1", 1)
[351110.029] wl_registry@15.global(8, "gtk_primary_selection_device_manager", 1)
[351110.056] wl_registry@15.global(9, "wl_subcompositor", 1)
[351110.080] wl_registry@15.global(10, "xdg_wm_base", 3)
[351110.107] wl_registry@15.global(11, "zxdg_shell_v6", 1)
[351110.134] wl_registry@15.global(12, "wl_shell", 1)
[351110.159] wl_registry@15.global(13, "gtk_shell1", 4)
[351110.184] wl_registry@15.global(14, "wp_viewporter", 1)
[351110.209] wl_registry@15.global(15, "zwp_pointer_gestures_v1", 1)
[351110.229] wl_registry@15.global(16, "zwp_tablet_manager_v2", 1)
[351110.257] wl_registry@15.global(17, "wl_seat", 5)
[351110.283] wl_registry@15.global(18, "zwp_relative_pointer_manager_v1", 1)
[351110.308] wl_registry@15.global(19, "zwp_pointer_constraints_v1", 1)
[351110.334] wl_registry@15.global(20, "zxdg_exporter_v1", 1)
[351110.359] wl_registry@15.global(21, "zxdg_importer_v1", 1)
[351110.385] wl_registry@15.global(22, "zwp_linux_dmabuf_v1", 3)
[351110.411]  -> wl_registry@15.bind(22, "zwp_linux_dmabuf_v1", 3, new id [unknown]@18)
[351110.445] wl_registry@15.global(23, "zwp_keyboard_shortcuts_inhibit_manager_v1", 1)
[351110.471] wl_registry@15.global(24, "zwp_text_input_manager_v3", 1)
[351110.499] wl_registry@15.global(25, "gtk_text_input_manager", 1)
[351110.525] wl_registry@15.global(26, "wp_presentation", 1)
[351110.551] wl_callback@16.done(3802)
[351110.568]  -> wl_display@1.sync(new id wl_callback@16)
[351111.975] wl_display@1.delete_id(16)
[351112.227] wl_drm@17.device("/dev/dri/renderD128")
[351112.433] wl_drm@17.format(808669761)
[351112.449] wl_drm@17.format(808669784)
[351112.463] wl_drm@17.format(808665665)
[351112.475] wl_drm@17.format(808665688)
[351112.487] wl_drm@17.format(875713089)
[351112.499] wl_drm@17.format(875713112)
[351112.513] wl_drm@17.format(909199186)
[351112.525] wl_drm@17.format(961959257)
[351112.536] wl_drm@17.format(825316697)
[351112.549] wl_drm@17.format(842093913)
[351112.560] wl_drm@17.format(909202777)
[351112.572] wl_drm@17.format(875713881)
[351112.583] wl_drm@17.format(842094158)
[351112.594] wl_drm@17.format(909203022)
[351112.605] wl_drm@17.format(1448695129)
[351112.617] wl_drm@17.capabilities(1)
[351112.628] zwp_linux_dmabuf_v1@18.format(875713089)
[351112.639] zwp_linux_dmabuf_v1@18.modifier(875713089, 16777215, 4294967295)
[351112.659] zwp_linux_dmabuf_v1@18.format(875713112)
[351112.670] zwp_linux_dmabuf_v1@18.modifier(875713112, 16777215, 4294967295)
[351112.691] zwp_linux_dmabuf_v1@18.format(808669761)
[351112.703] zwp_linux_dmabuf_v1@18.modifier(808669761, 16777215, 4294967295)
[351112.723] zwp_linux_dmabuf_v1@18.format(808665665)
[351112.735] zwp_linux_dmabuf_v1@18.modifier(808665665, 16777215, 4294967295)
[351112.757] zwp_linux_dmabuf_v1@18.format(808669784)
[351112.768] zwp_linux_dmabuf_v1@18.modifier(808669784, 16777215, 4294967295)
[351112.789] zwp_linux_dmabuf_v1@18.format(909199186)
[351112.802] zwp_linux_dmabuf_v1@18.modifier(909199186, 16777215, 4294967295)
[351112.840] zwp_linux_dmabuf_v1@18.format(1211384385)
[351112.869] zwp_linux_dmabuf_v1@18.modifier(1211384385, 16777215, 4294967295)
[351112.889] zwp_linux_dmabuf_v1@18.format(1211384408)
[351112.903] zwp_linux_dmabuf_v1@18.modifier(1211384408, 16777215, 4294967295)
[351112.923] zwp_linux_dmabuf_v1@18.format(1211388504)
[351112.935] zwp_linux_dmabuf_v1@18.modifier(1211388504, 16777215, 4294967295)
[351112.955] zwp_linux_dmabuf_v1@18.format(1211388481)
[351112.966] zwp_linux_dmabuf_v1@18.modifier(1211388481, 16777215, 4294967295)
[351112.988] wl_callback@16.done(3802)
[351113.006]  -> wl_display@1.sync(new id wl_callback@16)
[351114.867] wl_display@1.delete_id(16)
[351114.916] wl_callback@16.done(3802)
[351265.453]  -> wl_surface@5.frame(new id wl_callback@16)
[351266.080] wl_output@8.geometry(0, 0, 340, 190, 0, "CMN", "0x15db", 0)
[351266.168] wl_output@8.mode(3, 1366, 768, 60003)
[351266.204] wl_output@8.scale(1)
[351266.260] wl_output@8.done()
[351266.272] wl_seat@11.capabilities(3)
[351266.289]  -> wl_seat@11.get_pointer(new id wl_pointer@19)
[351266.309]  -> wl_seat@11.get_keyboard(new id wl_keyboard@20)
[351266.324] wp_presentation@12.clock_id(1)
[351266.350]  -> xdg_toplevel@13.set_title("[Kayoanime] Barakamon - S01E01 - mpv")
[351267.047] wl_keyboard@20.keymap(1, fd 20, 51300)
[351272.719]  -> wl_display@1.get_registry(new id wl_registry@21)
[351272.775]  -> wl_display@1.sync(new id wl_callback@22)
[351273.135] wl_display@1.delete_id(22)
[351273.158] wl_registry@21.global(1, "wl_drm", 2)
[351273.228]  -> wl_registry@21.bind(1, "wl_drm", 2, new id [unknown]@23)
[351273.274] wl_registry@21.global(2, "wl_compositor", 4)
[351273.296] wl_registry@21.global(3, "wl_shm", 1)
[351273.327] wl_registry@21.global(4, "wl_output", 2)
[351273.355] wl_registry@21.global(5, "zxdg_output_manager_v1", 3)
[351273.377] wl_registry@21.global(6, "wl_data_device_manager", 3)
[351273.397] wl_registry@21.global(7, "zwp_primary_selection_device_manager_v1", 1)
[351273.420] wl_registry@21.global(8, "gtk_primary_selection_device_manager", 1)
[351273.441] wl_registry@21.global(9, "wl_subcompositor", 1)
[351273.461] wl_registry@21.global(10, "xdg_wm_base", 3)
[351273.491] wl_registry@21.global(11, "zxdg_shell_v6", 1)
[351273.519] wl_registry@21.global(12, "wl_shell", 1)
[351273.546] wl_registry@21.global(13, "gtk_shell1", 4)
[351273.575] wl_registry@21.global(14, "wp_viewporter", 1)
[351273.602] wl_registry@21.global(15, "zwp_pointer_gestures_v1", 1)
[351273.621] wl_registry@21.global(16, "zwp_tablet_manager_v2", 1)
[351273.641] wl_registry@21.global(17, "wl_seat", 5)
[351273.669] wl_registry@21.global(18, "zwp_relative_pointer_manager_v1", 1)
[351273.690] wl_registry@21.global(19, "zwp_pointer_constraints_v1", 1)
[351273.710] wl_registry@21.global(20, "zxdg_exporter_v1", 1)
[351273.729] wl_registry@21.global(21, "zxdg_importer_v1", 1)
[351273.750] wl_registry@21.global(22, "zwp_linux_dmabuf_v1", 3)
[351273.777] wl_registry@21.global(23, "zwp_keyboard_shortcuts_inhibit_manager_v1", 1)
[351273.798] wl_registry@21.global(24, "zwp_text_input_manager_v3", 1)
[351273.828] wl_registry@21.global(25, "gtk_text_input_manager", 1)
[351273.858] wl_registry@21.global(26, "wp_presentation", 1)
[351273.887] wl_callback@22.done(3802)
[351273.903]  -> wl_display@1.sync(new id wl_callback@22)
[351274.150] wl_display@1.delete_id(22)
[351274.169] wl_drm@23.device("/dev/dri/renderD128")
[351274.305]  -> wl_drm@23.authenticate(0)
[351274.348] wl_drm@23.format(808669761)
[351274.365] wl_drm@23.format(808669784)
[351274.381] wl_drm@23.format(808665665)
[351274.397] wl_drm@23.format(808665688)
[351274.412] wl_drm@23.format(875713089)
[351274.425] wl_drm@23.format(875713112)
[351274.438] wl_drm@23.format(909199186)
[351274.451] wl_drm@23.format(961959257)
[351274.464] wl_drm@23.format(825316697)
[351274.477] wl_drm@23.format(842093913)
[351274.490] wl_drm@23.format(909202777)
[351274.503] wl_drm@23.format(875713881)
[351274.516] wl_drm@23.format(842094158)
[351274.528] wl_drm@23.format(909203022)
[351274.541] wl_drm@23.format(1448695129)
[351274.554] wl_drm@23.capabilities(1)
[351274.566] wl_callback@22.done(3802)
[351274.582]  -> wl_display@1.sync(new id wl_callback@22)
[351279.952] wl_display@1.error(wl_drm@23, 0, "authenticate failed")
wl_drm@23: error 0: authenticate failed
VA error: wayland: Wayland roundtrip error: Protocol error (errno 71)
[vaapi] libva: Open file 1 failed (Read-only file system)
[vaapi] libva: vaGetDriverNameByIndex() failed with invalid VADisplay, driver_name = (null)
[vo/gpu/wayland] Error occurred on the display fd, closing
```

Adding `--env=LIBVA_DRIVER_NAME=radeonsi` (in my case):

`[vaapi] libva: /usr/lib/x86_64-linux-gnu/GL/lib/dri/radeonsi_drv_video.so init failed`

This started happening recently I'll try to check if downgrading flatpak helps; at the moment flatpak seems to be the culprit.

**Comment by qvshuo** [comment]
same problem

**Comment by andreas-theo** [comment]
I also have the same problem. The Celluloid flatpak crashes immediately after running.

```
VA error: wayland: Wayland roundtrip error: Protocol error (errno 71)
Cannot load libcuda.so.1
Gdk-Message: 13:16:58.008: Error flushing display: Protocol error
```

**Comment by teohhanhui** [comment]
Try to check with

    $ sudo vainfo

If you get something like this:

```
libva info: Trying to open /usr/lib64/dri/radeonsi_drv_video.so
libva info: va_openDriver() returns -1
```

You might have a missing driver. See https://bugzilla.redhat.com/show_bug.cgi?id=1953769

For openSUSE you have to install the `Mesa-libva` package: https://pkgs.org/download/Mesa-libva
For Arch Linux it's `libva-mesa-driver`: https://pkgs.org/download/libva-mesa-driver

**Comment by andreas-theo** [comment]
Thanks for trying to help. I don't have an AMD GPU and I'm still affected by this bug so I don't think this is the problem. Although the OP might be affected by it.

```
libva info: VA-API version 1.11.0
libva info: Trying to open /usr/lib64/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_11
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.11 (libva 2.11.0)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 21.1.3 ()
vainfo: Supported profile and entrypoints
```
I didn't include the list of supported codecs.

**Comment by bbhtt** [comment]
@teohhanhui vainfo is working fine on host no missing driver here.

**Comment by teohhanhui** [comment]
`mpv` through flatpak works for me though (has it been fixed recently?). It's only `celluloid` that's not working...

**Comment by bbhtt** [comment]
> `mpv` through flatpak works for me though (has it been fixed recently?). It's only `celluloid` that's not working...

No, still same issue with `--hwdec=vaapi --gpu-context=wayland` here although these work on wayland:

1. `--hwdec=vaapi-copy --gpu-context=wayland`
2. ` --vo=gpu --gpu-context=wayland`
3. `flatpak run --env=LIBVA_DRIVER_NAME=<driver_name> io.mpv.Mpv --vo=vaapi --gpu-context=wayland` (doesn't work without the --env here)

It might not be a flatpak issue then. Can someone confirm these also work with celluloid and rest?

**Comment by andreas-theo** [comment]
I have tried the following with Celluloid and they don't seem to make a difference. Perhaps there is a better way to pass these to the mpv bundled in Celluloid:

1. `flatpak run io.github.celluloid_player.Celluloid --mpv-hwdec=vaapi-copy --mpv-gpu-context=wayland`
2. `flatpak run io.github.celluloid_player.Celluloid --mpv-vo=gpu --mpv-gpu-context=wayland`
3. `flatpak run --env=LIBVA_DRIVER_NAME=iHD_drv_video io.mpv.Mpv --vo=gpu --gpu-context=wayland`
Starting mpv with no arguments to enable hardware decoding works but then as soon as you press Ctrl-H it crashes. The terminal output is:
```
[mpv_thumbnail_script_client_osc]  
[mpv_thumbnail_script_client_osc] The 'tracks-changed' event is deprecated and will be removed.
 (+) Video --vid=1 (*) (vp9 3840x1920 29.970fps)
 (+) Audio --aid=1 (*) (aac 2ch 44100Hz)
[vo/gpu/wayland] GNOME's wayland compositor lacks support for the idle inhibit protocol. This means the screen can blank during playback.
AO: [pulse] 44100Hz stereo 2ch float
VO: [gpu] 3840x1920 yuv420p
wl_drm@31: error 0: authenticate failed0 Cache: 15s/18MB
VA error: wayland: Wayland roundtrip error: Protocol error (errno 71)

[vaapi] libva: vaGetDriverNameByIndex() failed with invalid VADisplay, driver_name = (null)
Cannot load libcuda.so.1
[ffmpeg] AVHWDeviceContext: Cannot load libcuda.so.1
[ffmpeg] AVHWDeviceContext: Could not dynamically load CUDA
AV: 00:00:01 / 00:14:20 (0%) A-V:  0.000 Cache: 16s/19MB
[vo/gpu/wayland] Error occurred on the display fd, closing
AV: 00:00:01 / 00:14:20 (0%) A-V:  0.000 Cache: 17s/22MB

Exiting... (Quit)
```

**Comment by stove-panini** [comment]
Fedora 34 with Mesa 21 here.

**Comment by andreas-theo** [comment]
Fedora Workstation 34 and Fedora Silverblue 34 with Mesa 21.1.1

**Comment by bbhtt** [comment]
> What distros and mesa versions is everyone here on? I'm not certain on the details yet, but it seems like Mesa 21.0 has some [changes regarding DRM authentication](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/9704), and the freedesktop runtime might need [this commit](https://github.com/intel/libva/pull/476) to fix it.
> 
> Note that this is an untested theory! But it's my best guess as to the issue.

Mesa 21.1.1. That might be it but the current libva is already updated so that should have it? https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/4820/diffs#26fddc9f07317f3d82b36fd1cd6c24130bdf8fb9_37_38

**Comment by teohhanhui** [comment]
> No, still same issue with --hwdec=vaapi --gpu-context=wayland

Oh, at least `mpv` works without using `hwdec=vaapi`

**Comment by xerz-one** [comment]
@refi64 Happens here trying to use Clapper on Arch Linux with Mesa 21.1.1 and Flatpak 1.11.1

**Comment by tinywrkb** [comment]
Same issue here with Intel Graphics (Broadwell) on Arch Linux running my own packaged MPV Flatpak app using Freedesktop 20.08 runtime.  
* Flatpak 1.11.1 <- No idea why the maintainer packaged an unstable release, maybe he likes Steam.
* Mesa 21.1.1-1
* libva 2.11.0
* libva-intel-driver-hybrid 2.4.1
* intel-hybrid-codec-driver 2.0.0.r169.edead0c

No issue when running on the host.
```
$ vainfo

/usr/lib/dri/hybrid_drv_video.so has no function __vaDriverInit_0_32
Not using hybrid_drv_video.so
vainfo: VA-API version: 1.11 (libva 2.11.1)
vainfo: Driver version: Intel i965 driver for Intel(R) Broadwell - 2.4.1
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Simple            :	VAEntrypointEncSlice
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointEncSlice
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileH264MultiviewHigh      :	VAEntrypointVLD
      VAProfileH264MultiviewHigh      :	VAEntrypointEncSlice
      VAProfileH264StereoHigh         :	VAEntrypointVLD
      VAProfileH264StereoHigh         :	VAEntrypointEncSlice
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD
      VAProfileNone                   :	VAEntrypointVideoProc
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileVP8Version0_3          :	VAEntrypointVLD
```


Running vainfo in Flatpak fails and it was working previously.
```
$ flatpak run --command=vainfo io.mpv.player

wl_drm@6: error 0: authenticate failed
VA error: wayland: Wayland roundtrip error: Protocol error (errno 71)
libva info: VA-API version 1.8.0
libva error: vaGetDriverNameByIndex() failed with invalid VADisplay, driver_name = (null)
vaInitialize failed with error code 3 (invalid VADisplay),exit
```

Bumping libva in and vaainfo in the sandbox fixes the issue
```
flatpak run --command=vainfo io.mpv.player
libva info: VA-API version 1.11.0
libva info: Trying to open /app/lib/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_11
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.11 (libva 2.11.1)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 21.1.3 ()
vainfo: Supported profile and entrypoints
      VAProfileNone                   :	VAEntrypointVideoProc
      VAProfileNone                   :	VAEntrypointStats
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Simple            :	VAEntrypointEncSlice
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointFEI
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointFEI
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264ConstrainedBaseline:	VAEntrypointFEI
      VAProfileVP8Version0_3          :	VAEntrypointVLD
``` 

```
$ flatpak run io.mpv.player  http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4

 (+) Video --vid=1 (*) (h264 1280x720 24.000fps)
 (+) Audio --aid=1 (*) (aac 2ch 44100Hz)
[ffmpeg] AVHWDeviceContext: Failed to query surface attributes: 20 (the requested function is not implemented).
[vo/gpu/vaapi-egl] failed to retrieve libavutil frame constraints
Cannot load libcuda.so.1
Using hardware decoding (vaapi).
[ao/pulse] The --audio-stream-silence option is set. This will break certain player behavior.
AO: [pulse] 44100Hz stereo 2ch float
VO: [gpu] 1280x720 vaapi[nv12]
AV: 00:00:01 / 00:09:56 (0%) A-V:  0.000 Cache: 352s/103MB
```

`libva-utils.json`

```
{
  "name": "libva-utils",
  "buildsystem": "meson",
  "sources": [
    {
      "type": "archive",
      "url": "https://github.com/intel/libva-utils/archive/2.11.1.tar.gz",
      "sha256": "0c1eb7f717e391d00da74c53a9fe5caf3d6c510dcd35bac7f71a0e59ad1b8d26"
    }
  ],
  "modules": [
    {
      "name": "libva",
      "buildsystem": "meson",
      "sources": [
        {
          "type": "git",
          "url": "https://github.com/intel/libva.git",
          "tag": "64d75c08a720b73f4474b9265ca6dc99f947af8b"
        }
      ]
    },
    {
      "name": "intel-media-driver",
      "buildsystem": "cmake",
      "builddir": true,
      "sources": [
        {
          "type": "archive",
          "url": "https://github.com/intel/media-driver/archive/intel-media-21.1.3.tar.gz",
          "sha256": "219ce6b08a84bdce311160dc694d866249fd4e390391c2ac7be55f13a2fb928c"
        }
      ],
      "modules": [
        {
          "name": "intel-gmmlib",
          "buildsystem": "cmake",
          "builddir": true,
          "sources": [
            {
              "type": "archive",
              "url": "https://github.com/intel/gmmlib/archive/intel-gmmlib-21.1.3.tar.gz",
              "sha256": "c459a95243075296c72a40674c449ffcc314ed6d21ad0291c1603dbc11f4a74e"
            }
          ]
        }
      ]
    }
  ]
}
```

**Comment by tinywrkb** [comment]
Oh, we need to wait for 20.08.13.
```
$ flatpak run --system --command=cat org.freedesktop.Platform//20.08 /etc/issue
Freedesktop Platform 20.08.12 \n \l
```
I'll test again later when the update will be available.

**Comment by bbhtt** [comment]
From my testing MPV and Celluloid works again with `org.freedesktop.Platform 20.08.13`.

**Comment by tinywrkb** [comment]
I can also confirm that the issue is solved now with 20.08.13.

**Comment by andreas-theo** [comment]
The issue has been solved with MPV and Celluloid. Thank you @refi64 🙏
